android/src/main/java/io/ably/flutter/plugin/AblyMessageCodec.java

```diff
@@ -356,25 +369,31 @@ private TokenDetails decodeTokenDetails(Map<String, Object> jsonMap) {
 
   private ChannelOptions decodeRestChannelOptions(Map<String, Object> jsonMap) {
     if (jsonMap == null) return null;
-    final Object cipher = jsonMap.get(PlatformConstants.TxRealtimeChannelOptions.cipher);
-    try {
-      return createChannelOptions(cipher);
-    } catch (AblyException e) {
-      System.out.println("Exception while decoding RestChannelOptions: " + e);
-      return null;
+    ChannelOptions options = new ChannelOptions();
+    options.cipherParams = decodeCipherParams((Map<String, Object>) jsonMap.get(PlatformConstants.TxRestChannelOptions.cipherParams));
+    if (options.cipherParams != null) {
+      options.encrypted = true;
     }
+    return options;
+  }
+
+  /// RealtimeChannelOptions (the inheriting class) is used for both RealtimeChannelOptions and
+  // RestChannelOptions, since Ably Java has only one type: ChannelOptions
+  private Map<String, Object> encodeChannelOptions(final ChannelOptions c) {
+    if (c == null) return null;
```

Comment:
I'm seen this pattern before in this codebase, and may have commented on it then, but not sure if that was similar context. How come this contract can be so loose? Surely `null` here _should_ throw [NPE](https://docs.oracle.com/javase/7/docs/api/java/lang/NullPointerException.html) or otherwise fail hard?